modules/internal/ChapelRange.chpl

```diff
@@ -617,27 +618,28 @@ module ChapelRange {
     if chpl__singleValIdxType(idxType) {
       if _low > _high { // avoid circularity of calling .size which calls .high
         warning("This range is empty and has a single-value idxType, so its high bound isn't trustworthy");
-        return this.alignedLow;
+        return this.low;
+      }
+    }
+    if chpl__singleValIdxType(idxType) {
+      if size == 0 {
+        warning("This range is empty and has a single-value idxType, so its high bound isn't trustworthy");
+        return chpl_intToIdx(_low);
       }
     }
-    return this.alignedHigh;
+    return chpl_intToIdx(this.alignedHighAsInt);
   }
 
 
   /* Return the range's aligned high bound.  Note that this is a
      synonym for :proc:`range.high`.
   */
+  deprecated "'.alignedHigh' is deprecated; please use '.high' instead"
   inline proc range.alignedHigh: idxType {
     if !hasHighBound() {
       compilerError("can't query the high bound of a range without one");
     }
-    if chpl__singleValIdxType(idxType) {
```

Comment:
Since this routine now calls `range.high`, it inherits the check and would generate two warnings if it were in both routines.  So it's less that it's being removed from one place while being added to another, and more that it's being moved between routines as they switch roles (from "I call you and you implement the real thing" to "You call me and I implement it").

Reply:
It definitely could be, but unlike the previous case we were discussing, there's also no harm in having it in both places since it's a (compile-time) error rather than a warning (so won't get generated twice).  Sometimes we haven't been good at having compiler errors print the user code's callsite when there have been chains of calls between the user code and the error, which _could_ be a reason to duplicate it, but that doesn't seem to currently be the case for this case.

If this were code we were maintaining long-term, I'd be more inclined to remove the duplication to ensure the error messages remained consistent and to reduce maintenance costs going forward.  But since it's in a routine that will be going away as soon as we can get rid of it, it feels a bit like busy-work to me to remove it and re-run testing, particularly since it's a pre-existing condition.  Let me know if you disagree, though.

Reply:
I understand those distinctions and I agree.